src/main/java/org/opensearch/knn/training/VectorReader.java

```diff
@@ -89,7 +89,7 @@ public void read(
             throw validationException;
         }
 
-        ValidationException fieldValidationException = IndexUtil.validateKnnField(indexMetadata, fieldName, -1, null);
+        ValidationException fieldValidationException = IndexUtil.validateKnnField(indexMetadata, fieldName, -1, null, null);
```

Comment:
Why is this null at the end?

Reply:
I think the existing `IndexUtil.validateKnnField` function is shared for both index field and model field validation, in above case in the vectorReader, it's used in case of index field validation, similarly like the fourth parameter of `modalDao`, I pass expectedVectorDataType with null.

In ModelDaoRequest class, we use it as:
```
        ValidationException fieldValidation = IndexUtil.validateKnnField(
            indexMetadata,
            this.trainingField,
            this.dimension,
            modelDao,
            this.vectorDataType
        );
```